suite/suite.go

```diff
@@ -184,6 +184,11 @@ func Run(t *testing.T, suite TestingSuite) {
 					failOnPanic(t, r)
 				}()
 
+				if method.Type.NumIn() > 1 || method.Type.NumOut() > 0 {
+					msg := fmt.Sprintf("testify: suite method '%s' shouldn't have any arguments or returning values\n", method.Name)
+					panic(msg)
```

Comment:
Not sure about using panic here tho, seems like appropriate, but will appreciate if you take a look @dolmen

Reply:
Isn't it better to use t.Error(msg)? Then you can directly see the TestMethod violating the contract. I find panic always quite hard.

Reply:
Yeah fair point, I just hate reading panics.

Thanks for putting in the work.

Reply:
While I'm a strong advocate of `panic` to catch programmer's mistake, I think a `panic` is not appropriate in this case as the stack trace will not be useful to the user.

Instead we are being called with `*testing.T`, so let's just use `t.Fatal` (or `t.Fatalf`) to report the issue.

However, the main problem with this implementation is that this check must be done much much earlier, line 144, just after the successful call to `methodFilter`.